### PR TITLE
Restructure pipeline

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/CenterForMedicalGeneticsGhent/nf-cmgg-germline/blob/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by nvnieuwk** [comment]
@matthdsm ping

File: nextflow.config [hunk 1]
```diff
@@ -21,6 +21,8 @@ params {
     use_dragstr_model          = false
     skip_genotyping            = false
     use_bcftools_merge         = false
+    dbsnp                      = null
```
Comment (matthdsm) [inline nextflow.config:24]: Also add dbsnp to the genomes/igenomes config, so it gets auto detected

File: workflows/cmgg-germline.nf [hunk 2]
```diff
@@ -37,6 +39,10 @@ if (params.input) { ch_input = file(params.input, checkIfExists: true) } else {
 // Check for dependencies between parameters
 //
 
+if((params.dbsnp || params.dbnsp_tbi) && !(params.dbsnp && params.dbsnp_tbi)){
+    exit 1, "You only specified --dbsnp or --dbsnp_tbi. Please specify both arguments with the correct file inputs."
```
Comment (matthdsm) [inline workflows/cmgg-germline.nf:43]: Tabix when index is not available?

File: conf/genomes.config [hunk 3]
```diff
@@ -15,6 +15,8 @@ params {
             fai          = "${params.genomes_base}/Hsapiens/hg38/seq/hg38.fa.fai"
             fasta        = "${params.genomes_base}/Hsapiens/hg38/seq/hg38.fa"
             strtablefile = "${params.genomes_base}/Hsapiens/hg38/variation/hg38.str.zip"
+            dbsnp        = "${params.genomes_base}/Hsapiens/hg38/variation/dbsnp-153.vcf.gz"
```
Comment (matthdsm) [inline conf/genomes.config:18]: dbsnp version in our references store is v151

File: conf/genomes.config [hunk 4]
```diff
@@ -15,6 +15,8 @@ params {
...
+            dbsnp_tbi    = "${params.genomes_base}/Hsapiens/hg38/variation/dbsnp-153.vcf.gz.tbi"
```
Comment (matthdsm) [inline conf/genomes.config:19]:
```suggestion
            dbsnp_tbi    = "${params.genomes_base}/Hsapiens/hg38/variation/dbsnp-153.vcf.gz.tbi"
            vep_cache   = "${params.genomes_base}/Hsapiens/hg38/vep"
```

File: conf/genomes.config (hunk 3 above)
Comment (matthdsm) [reply conf/genomes.config:18]:
```suggestion
            dbsnp        = "${params.genomes_base}/Hsapiens/hg38/variation/dbsnp-151.vcf.gz"
```

File: conf/genomes.config (hunk 4 above)
Comment (matthdsm) [inline conf/genomes.config:19]:
```suggestion
            dbsnp_tbi    = "${params.genomes_base}/Hsapiens/hg38/variation/dbsnp-151.vcf.gz.tbi"
```

File: assets/vcfanno.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # the resources in this file and directory are taken from https://github.com/brentp/vcfanno/blob/master/example
 [[annotation]]
-file="exac.vcf.gz"
+file="variation/exac.vcf.gz"
```
Comment (matthdsm) [inline assets/vcfanno.toml:3]: Update the module first, I fixed that

Comment (nvnieuwk) [reply assets/vcfanno.toml:3]: I updated the module in the last commit, it wasn't fixed...

Comment (nvnieuwk) [reply assets/vcfanno.toml:3]: But it works now using this TOML

Comment (matthdsm) [reply assets/vcfanno.toml:3]: because the resource bundle wasn't fixed, the module and test was

Comment (nvnieuwk) [reply assets/vcfanno.toml:3]: Yeah the fixes are already in the pipeline :)

**Comment by matthdsm** [comment]
What's the error from vcf2db?

**Comment by nvnieuwk** [comment]
Well sometimes the pipeline runs VCF2DB and sometimes it doesn't, probably some channel mixups somewhere. I'll try and fix it today

**Comment by nvnieuwk** [comment]
Linting tests will keep failing for now. This will be fixed in v2.8 of `nf-core`

**Pull request merged**